src/go/rpk/pkg/cli/cmd/wasm/deploy.go

```diff
@@ -0,0 +1,106 @@
+package transform
+
+import (
+	"encoding/binary"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"vectorized/pkg/utils"
+
+	"github.com/Shopify/sarama"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func NewDeployCommand(
+	fs afero.Fs, createProduce func(bool, int32) (sarama.SyncProducer, error),
+) *cobra.Command {
+	var (
+		description	string
+		jvmPartitioner	bool
+		partition	int32
+	)
+
+	command := &cobra.Command{
+		Use:	"deploy <path>",
+		Short:	"deploy inline WASM function",
+		Args:	cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			path := args[0]
+			fullFileName := filepath.Base(path)
+			fileExt := filepath.Ext(fullFileName)
+			// validate file extension, just allow js extension
+			if fileExt != ".js" {
+				return fmt.Errorf("deploy just allow js extension file")
+			}
+			fileName := strings.Replace(fullFileName, fileExt, "", 1)
+			fileContent, err := utils.ReadFileLinesBinary(fs, path)
+			if err != nil {
+				return err
+			}
+			// create producer
+			producer, err := createProduce(jvmPartitioner, partition)
+
+			if err != nil {
+				return err
+			}
+			return deploy(
+				fileName,
+				fileContent,
+				description,
+				producer,
+			)
+		},
+	}
+
+	command.Flags().StringVar(
+		&description,
+		"description",
+		"",
+		"wasm definition description",
+	)
+
+	command.Flags().BoolVarP(
+		&jvmPartitioner,
+		"jvm-partitioner",
+		"j",
+		false,
+		"Use a JVM-compatible partitioner. If --partition is passed"+
+			" with a positive value, this will be overridden and"+
+			" a manual partitioner will be used.",
+	)
+
+	command.Flags().Int32VarP(
+		&partition,
+		"partition",
+		"p",
+		-1,
+		"Partition to produce to.",
+	)
+
+	return command
+}
+
+func deploy(
+	fileName string,
+	fileContent []byte,
+	description string,
+	producer sarama.SyncProducer,
+) error {
+	// create buffer for filename and content size bytes
+	sizeFileName := make([]byte, 16)
+	sizeFileContent := make([]byte, 64)
+	sizeDescription := make([]byte, 64)
+	binary.LittleEndian.PutUint16(sizeFileName, uint16(len([]byte(fileName))))
+	binary.LittleEndian.PutUint64(sizeFileContent, uint64(len(fileContent)))
+	binary.LittleEndian.PutUint64(sizeDescription, uint64(len(description)))
+	// join all bites on one buffer
+	messageContent := append(sizeFileName, []byte(fileName)...)
```

Comment:
we don't have the coproc ID into the file content, the coprocessor Id is the filename in a hash64 format, for that reason we need to put that name into the topic, but after reading this comment, I think we can get the id from the key record, `deploy/name` 
is it related to your question? also we want to add a description in order to give the user some information about the specific coprocessor.

Reply:
Yeah it answers the question, nice solution. How would the user ever see the description? Its not like they should be allowed to read from this topic, redpanda will read from the topic and then deploy the scripts. I think it would be better for these attributes (like name/description) to be in the coproc metadata anyway. Because if its there, it can be later queried on demand.

Reply:
Yes, the user doesn't read the internal topic, but the `rpk wasm status` can read from that topic and show the description for each coprocessor, we had a name and description as metadata on coprocessor definition but we changed it because those attribute doesn't part of the coproc requirements

Reply:
ah ok thats pretty cool. Lets stick with it then, we don't want to change the wasm engine as well if the file name is integral to how it works